app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -115,7 +116,9 @@ class FirestoreDatabaseService(
         if (!ref.get().await().exists()) {
             throw java.lang.IllegalArgumentException("The database ${this.db} does not contain the category ${category.id}")
         }
-        ref.delete().addOnFailureListener {
+        try {
+            ref.delete()
+        } catch (e: Exception) {
             Log.w(
                 this.toString(),
                 "Removing category ${category.id} from ${this.db} failed"
```

Comment:
```suggestion
                "Removing category ${category.id} from ${this.db} failed",
                e
```
If we're logging it's worth including the exception